### PR TITLE
Chapter 10

### PR DESCRIPTION
**Pull request by mohdzakwan99**

### CODE REVIEW

**Comment by mohdzakwan99** [comment]
ready

File: config/routes.rb [hunk 1]
```diff
@@ -1,11 +1,17 @@
...
 
   get "/signup", to: "users#new"
   post "/signup", to: "users#create"
-  resources :users, only: %i(new create show)
+
   get "/login", to: "sessions#new"
   post "/login", to: "sessions#create"
   delete "/logout", to: "sessions#destroy"
+
+  get "/edit", to: "users#edit"
```
Comment (huongnt-2545) [inline config/routes.rb:12]: remove line 12 -> 15

File: config/routes.rb [hunk 2]
```diff
@@ -1,11 +1,17 @@
...
+  post "/edit", to: "users#update"
+
+  get "/users", to: "users#index"
+  resources :users, only: %i(new create show destroy edit update)
```
Comment (huongnt-2545) [inline config/routes.rb:16]:
```suggestion
  resources :users
```

File: test/controllers/users_controller_test.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+require "test_helper"
```
Comment (huongnt-2545) [inline test/controllers/users_controller_test.rb:1]: delete this file

**Comment by mohdzakwan99** [comment]
ready

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -1,4 +1,12 @@
 class UsersController < ApplicationController
+  before_action :logged_in_user, except: [:index, :edit, :update, :destroy]
```
Comment (anlvq-1058) [inline app/controllers/users_controller.rb:2]: Please fix your line 2

**Review by anlvq-1058 — COMMENTED** [review]
cmt

**Comment by mohdzakwan99** [comment]
ready

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -13,21 +21,57 @@ def create
...
   end
 
   private
 
+  def logged_in_user
+    unless logged_in?
+      store_location
+      flash[:danger] = "Please log in."
+      redirect_to login_url
+    end
```
Comment (anlvq-1058) [inline app/controllers/users_controller.rb:67]:
```suggestion
    return if logged_in?
    
    store_location
    flash[:danger] = "Please log in."
    redirect_to login_url
```

File: app/views/users/index.html.erb [hunk 6]
```diff
@@ -0,0 +1,9 @@
+<% provide(:title, "All users") %>
+<h1>All users</h1>
+<%= will_paginate @users%>
+<ul class="users">
+  <% @users.each do |user| %>
+    <%= render user %>
+  <% end %>
+</ul>
+<%= will_paginate @users%>
```
Comment (anlvq-1058) [inline app/views/users/index.html.erb:9]:
```suggestion
<%= will_paginate @users %>
```

File: app/controllers/users_controller.rb [hunk 7]
```diff
@@ -13,21 +21,57 @@ def create
...
+    redirect_to users_url
   end
 
   private
 
+  def logged_in_user
+    unless logged_in?
+      store_location
+      flash[:danger] = "Please log in."
+      redirect_to login_url
```
Comment (anlvq-1058) [inline app/controllers/users_controller.rb:66]: align line 64, 65, 66 with unless

**Review by anlvq-1058 — COMMENTED** [review]
cmt

**Comment by mohdzakwan99** [comment]
ready

**Comment by ngocvt-0484** [comment]
merged

**Pull request merged**